sorts/binary_search.cpp

```diff
@@ -0,0 +1,51 @@
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+const int32_t binarySearch(const int32_t value, const std::vector<int32_t> vector, const size_t size) {
+    if (size < 0) {
+        throw std::invalid_argument("Given argument [size] is invalid - cannot be less than minimal treshold");
+    }
+
+    uint32_t midPoint = size / 2;
```

Comment:
Read about iterators and iterator arithmetic, then apply this knowledge and get rid of `size` parameter